README.md

```diff
@@ -144,18 +173,18 @@ this file. If I need to do anything more that type a single command to run your
 code, include a script or makefile that will do the job.
 
 ## A2.5
+See grammar_Ans2.cpp
```

Comment:
This code was failing a few tests. When I looked into it, I found that your state machine code was using incorrect.

The state machine should be taking the words, one at a time, and updating the state. But your original code was using a single variable (`current`) as both the state and the offset to the next word. It was also not checking that it had reached the end state at the end of the sentence, so it would have allowed "the dog ate slowly as the sun set fast in the west."

Here's a corrected version

~~~ c++
bool trySentence(vector<string> words){
  map<int, map<string, int>> states = {
    {0, {{"the", 1}}},
    {1, {{"smelly", 2}, {"lazy", 2}, {"dog", 3}, {"cat", 3}}},
    {2, {{"smelly", 2}, {"lazy", 2}, {"dog", 3}, {"cat", 3}}},
    {3, {{"ate", 4}, {"ran", 4}}},
    {4, {{"slowly", 5}, {"noisily", 5},{"EOI", 99}}},
    {5, {{"EOI", 99}}}
  };
  int state = 0;
  int current_word = 0;
  while (current_word < words.size()){
    state = NextState(state, words[current_word++], states);
    if (state < 0)
      return false;
  }
  return state == 99;
}
~~~

I also can't believe how labor intensive writing those test cases was. I'm giving you some extra points just in sympathy.

# -6